src/iam/iam-base.ts

```diff
@@ -165,6 +166,13 @@ export class IAMBase {
         this.setJWT();
     }
 
+    private async checkBalanceForConnection() {
+        const balance = await this._signer?.provider?.getBalance(await this._signer?.getAddress());
+        if (balance?.toString() === "0") {
```

Comment:
It is not enough to ensure that transaction fee will be paid. Balance should be compared with estimated pay of the transaction

Reply:
Updated code to calculate fee like in staking.